keps/74-support-argo-workflow/README.md

```diff
@@ -0,0 +1,350 @@
+    - [Examples](#examples)
+      - [Example 1 (ParallelSteps Contains Leaf Template Only)](#example-1-parallelsteps-contains-leaf-template-only)
+      - [Example 2 (ParallelSteps Contains Leaf Template and Step)](#example-2-parallelsteps-contains-leaf-template-and-step)
+      - [Example 3 (Workflow with Single Container Template)](#example-3-workflow-with-single-container-template)
+    - [How to suspend a workflow step by step](#how-to-suspend-a-workflow-step-by-step)
+    - [Drawback and Limitations](#drawback-and-limitations-1)
+    - [Advantages](#advantages-1)
+  - [Plain Pod as An Unit](#plain-pod-as-an-unit)
+    - [Drawback and Limitations](#drawback-and-limitations-2)
+    - [Advantages](#advantages-2)
+- [Additional Details](#additional-details)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This KEP outlines the proposal to integrate Argo Workflows within Kueue, discussing the advantages 
+and disadvantages of queuing workflows at varying granularity levels, alongside detailing 
+the integration methodologies.
+
+## Motivation
+
+Workflows are pivotal components in domains like Scientific Computing and Simulations, where 
+administrators often enforce resource usage quotas for different users or departments. Currently, 
+Argo Workflows lacks native support within Kueue.
+
+### Goals
+
+- Enable support for Argo Workflow within Kueue, allowing users to simply add a label 
+`kueue.x-k8s.io/queue-name` to their workflows and submit them initially in a suspended state.
+- Should be easily extended to support other workflow managers.
+
+### Non-Goals
+
+
+
+## Proposal
+
+
+### User Stories
+
+#### Story 1
+
+As a machine learning engineer, I need to preprocess data before executing a training job. My 
+workflow includes two steps: data preprocessing (which doesn't require a GPU) followed by a 
+PyTorchJob. I desire that the data preprocessing stage proceeds independently of GPU quota 
+availability.
+
+#### Story 2
+
+As an ML engineer, my workflow consists of several GPU-dependent stages with uniform resource 
+requirements. I aim to recycle resources allocated to earlier workflow stages to boost efficiency 
+and resource utilization.
+
+## Design Details
+
+### Workflow as An Unit
+
+Given the diverse resource, node affinity, and toleration requirements among workflow pods, 
+determining the necessary resources for each flavor becomes challenging for the controller. 
+Users must specify workflow resources via annotations like kueue.k8s.io/max-resources, tolerations 
+with kueue.k8s.io/toleration, and node selectors with kueue.k8s.io/node-selector.
+
+#### Drawback and Limitations
+
+- Inability to set distinct nodeSelectors and tolerations for multiple pod sets within a workflow.
```

Comment:
It also wouldn't be obvious how to inject any node selectors that Kueue could calculate from flavor assignment.